docs/project/requirements/architecture.md

```diff
@@ -0,0 +1,147 @@
+# Architecture
+
+Un schéma d'implémentation détaillé sera établi pour décrire l'architecture globale de l'infrastructure, y compris les différents composants, leur interaction et leur déploiement.
+
+
+## Application
+
+
+### Proxy: Traefik
+
+* Application router
+* High-availability by fault tolerance
+* Load-balancing by requests distribution
+  ALB ?
+* Helm chart
+* Version
+
+
+### Application: FastAPI
+
+* Python
+  * Version:
+  * Packaging
+* Project management? Poetry?
+
+
+### Databases: PostgreSQL
+
+* Helm chart
+* Version
+* Master/slave for high availability and read load balancing
+* Master failover
+
+
+## Infrastructure
+
+
+### Network
+
+
+#### Application Load Balancer (ALB)
+
+
+#### Virtual Private Cloud (VPC)
+
+* VPC subnet
+
+
+#### Internet Gateways (IGW)
+
+* Global egress
+
+
+#### Availability Zones (AZ)
+
+* 2 availability zones for high availability
+
+
+#### NAT Gateways (NGW)
+
+* Per AZ egress
+* In public subnet
+
+
+#### Subnets
+
+
+##### Public subnets
+
+* Application access
+* Bastion access
+* NAT gateways ?
+
+
+##### Private subnets
+
+* Database
+* Admin
+
+
+### Compute
+
+#### Nodes
+
+* EC2 compute nodes
+  * Application nodes
+  * Database nodes
+  * Bastion node
+* Auto-scaling groups
+
+
+#### Orchestration: Kubernetes
+
+#### Pods
+
+##### ReplicaSets
```

Comment:
Do we need this granularity? The HELM charts manage the type of deployment automatically.
We can look it up, if we want to document it.

Reply:
We, at least, need to explicit our replication strategy and implementation principle, be it delegation to Helm (and operator?) with appropriate parameter values.

Reply:
The replication strategy deserves its own section or even its own file, IMHO. There's a lot to say on that front.
Eg. we need to add notes on how our apps are deployed, how our databases are managed, etc.